src/Backend/Modules/Mailmotor/Actions/Ping.php

```diff
@@ -24,14 +25,38 @@ public function execute(): void
         parent::execute();
 
         // Successful API connection
-        if ($this->get('mailmotor.subscriber')->ping()) {
+        if ($this->ping()) {
             $this->redirect($this->getBackLink(['report' => 'successful-mail-engine-api-connection']));
         }
 
         $this->resetMailEngine();
         $this->redirect($this->getBackLink(['error' => 'wrong-mail-engine-credentials']));
     }
 
+    private function ping(): bool
+    {
+        $gateway = $this->getContainer()->get('mailmotor.factory')->getSubscriberGateway();
+
+        if (!$gateway->ping($this->getContainer()->getParameter('mailmotor.list_id'))) {
```

Comment:
Could you add some comments, because I have no clue what this method does. I find the name `ping` not really descriptive. I guess it checks if some list exists?

Reply:
fixed in a7352acf1

Reply:
clarified in 4cf05dc